storm-core/test/jvm/org/apache/storm/MessagingTest.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm;
+
+import org.apache.storm.generated.GlobalStreamId;
+import org.apache.storm.generated.Grouping;
+import org.apache.storm.generated.StormTopology;
+import org.apache.storm.testing.*;
+import org.apache.storm.utils.Utils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class MessagingTest {
+    private final static Logger LOG = LoggerFactory.getLogger(MessagingTest.class);
+
+    @Test
+    public void testLocalTransport() throws Exception {
+        Config stormConf = new Config();
+        //stormConf.putAll(Utils.readDefaultConfig());
+        stormConf.put(Config.TOPOLOGY_WORKERS, 2);
+        stormConf.put(Config.STORM_MESSAGING_TRANSPORT , "org.apache.storm.messaging.netty.Context");
```

Comment:
This was in the original code https://github.com/apache/storm/blob/d5acec9e3b9473a0e8cf39c7e12393626a3ca426/storm-core/test/clj/org/apache/storm/messaging_test.clj#L32-L33

But yes it should be optional.